### ISSUE TITLE
.has(/* nothing */) should be true

### ISSUE THREAD

**Issue by fiatjaf** [mariocasciaro/object-path#63]
code on https://github.com/mariocasciaro/object-path/blob/c540af3b3caea2cd776fb3c4afaa68448e186d99/index.js#L160-L162 for the `.has()` method:
```javascript
    if (isEmpty(path) || path.length === 0) {
      return false;
    }
```

If we are searching for an empty key and the variable being searched is an object, then it is intuitive that the object has that. It can perhaps miss all keys in the world as long as long as they have a name, but for keys with no name any object has all.

**Comment by fiatjaf** [comment]
In fact, it is the same rationale for `.get`, which is implemented the intuitive way ([code](https://github.com/mariocasciaro/object-path/blob/c540af3b3caea2cd776fb3c4afaa68448e186d99/index.js#L252-L254)).

For example:

```javascript
var a = {x: 123}

objectPath.get(a, 'x') // 123
objectPath.has(a, 'x') // true

objectPath.get(a) // {x: 123}
objectPath.has(a) // false -- why?
```

Do you agree this is a discrepancy?

**Comment by mariocasciaro** [comment]
I agree with you, there is a discrepancy between the two methods which might easily result in defects in the code. We are currently evaluating if the fix should be in the `get` method rather than in the `has` method. Please join the discussion here https://github.com/mariocasciaro/object-path/issues/62
